scapy/arch/linux.py

```diff
@@ -359,9 +359,9 @@ def get_last_packet_timestamp(sock):
 
 
 def _flush_fd(fd):
-    if type(fd) is not int:
+    if not isinstance(fd, int):
```

Comment:
We may want to change this to something like: `if hasattr(fd, 'fileno'):`. What do you think?